src/main/java/org/craftercms/studio/impl/v2/service/aws/mediaconvert/AwsMediaConvertServiceImpl.java

```diff
@@ -100,33 +100,19 @@ public class AwsMediaConvertServiceImpl extends AbstractAwsService<MediaConvertP
      */
     protected String smoothExtension;
 
-    public void setPartSize(final int partSize) {
-        this.partSize = partSize;
-    }
-
-    @Required
-    public void setDelimiter(final String delimiter) {
+    public AwsMediaConvertServiceImpl(SiteAwareConfigProfileLoader<MediaConvertProfile> profileLoader,
+                                      final String delimiter, final String urlPattern, final String hlsExtension,
+                                      final String dashExtension, final String smoothExtension) {
+        super(profileLoader);
         this.delimiter = delimiter;
-    }
-
-    @Required
-    public void setUrlPattern(final String urlPattern) {
         this.urlPattern = urlPattern;
-    }
-
-    @Required
-    public void setHlsExtension(final String hlsExtension) {
         this.hlsExtension = hlsExtension;
-    }
-
-    @Required
-    public void setDashExtension(final String dashExtension) {
         this.dashExtension = dashExtension;
+        this.smoothExtension = smoothExtension;
     }
 
-    @Required
-    public void setSmoothExtension(final String smoothExtension) {
-        this.smoothExtension = smoothExtension;
+    public void setPartSize(final int partSize) {
```

Comment:
Could you let me know if you are keeping this outside the constructor because it is used that way?

Reply:
@sumerjabri Same reason for the Engine. We only migrate setter with the annotation `@Required`. For this particular setter, we don't use it at all: https://github.com/craftercms/studio/blob/develop/src/main/resources/crafter/studio/studio-services-context.xml#L1273 in the bean initialization